sdk/storage/inc/datalake/protocol/datalake_rest_client.hpp

```diff
@@ -0,0 +1,2220 @@
+      result.RequestId = response.GetHeaders().at(k_HeaderXMsRequestId);
+      result.Version = response.GetHeaders().at(k_HeaderXMsVersion);
+      result.SetAccessControlRecursiveResponseObject = SetAccessControlRecursiveResponse::CreateFromJson(nlohmann::json::parse(response.GetBodyBuffer()));
+    }
+    else if (response.GetStatusCode() == Azure::Core::Http::HttpStatusCode::Accepted)
+    {
+      // The uploaded data was accepted.
+      result.ContentMD5 = response.GetHeaders().at(k_HeaderContentMD5);
+      result.Date = response.GetHeaders().at(k_HeaderDate);
+      result.RequestId = response.GetHeaders().at(k_HeaderXMsRequestId);
+      result.Version = response.GetHeaders().at(k_HeaderXMsVersion);
+    }
+    else
+    {
+      std::cout << "Error occured:\n"
+                << std::string(response.GetBodyBuffer().begin(), response.GetBodyBuffer().end()) << std::endl;
+      // throw StorageError::CreateFromXml(XmlWrapper::CreateFromBodyBuffer(response.GetBodyBuffer()));
+    }
+    return result;
+  }
+
+  static PathUpdateResponse PathUpdate(std::string url, std::shared_ptr<Azure::Core::Http::HttpTransport> transport, Azure::Core::Context& context, const PathUpdateOptions& pathUpdateOptions)
+  {
+    // TODO: Pipeline will be added when available.
+    return PathUpdateParseResponse(transport->Send(context, PathUpdateCreateRequest(std::move(url), pathUpdateOptions)));
+  }
+
+  struct PathLeaseOptions
+  {
+    std::string ClientRequestId = std::string();  // Provides a client-generated, opaque value with a 1 KB character limit that is recorded in the analytics logs when storage analytics logging is enabled.
+    uint32_t Timeout = uint32_t();  // The timeout parameter is expressed in seconds. For more information, see <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/fileservices/setting-timeouts-for-blob-service-operations">Setting Timeouts for Blob Service Operations.</a>
+    std::string ApiVersionParameter = k_DefaultServiceApiVersion;  // Specifies the version of the operation to use for this request.
+    PathLeaseAction XMsLeaseAction;  // There are five lease actions: "acquire", "break", "change", "renew", and "release". Use "acquire" and specify the "x-ms-proposed-lease-id" and "x-ms-lease-duration" to acquire a new lease. Use "break" to break an existing lease. When a lease is broken, the lease break period is allowed to elapse, during which time no lease operation except break and release can be performed on the file. When a lease is successfully broken, the response indicates the interval in seconds until a new lease can be acquired. Use "change" and specify the current lease ID in "x-ms-lease-id" and the new lease ID in "x-ms-proposed-lease-id" to change the lease ID of an active lease. Use "renew" and specify the "x-ms-lease-id" to renew an existing lease. Use "release" and specify the "x-ms-lease-id" to release a lease.
```

Comment:
can't we name it `LeaseAction`?

Reply:
Defined by swagger, I don't think customer would care.

Reply:
In js sdk they name it `leaseID` [link](https://github.com/Azure/azure-sdk-for-js/blob/a195ac041a26ec26802b113db74e6825c6fc8338/sdk/storage/storage-blob/src/generated/src/operations/blob.ts#L305). In .Net SDK, they also name it `leaseId` [this file](https://raw.githubusercontent.com/Azure/azure-sdk-for-net/master/sdk/storage/Azure.Storage.Blobs/src/Generated/BlobRestClient.cs) on line 9396.

Both are generated code, but they give it a reasonable name. I don't think the code being auto-generated means we can feel free to name the variables.

Reply:
fine.